pkg/solver/matcher/matcher.go

```diff
@@ -136,49 +131,74 @@ func GetMatchingDeals(
-				}))
-			deal, err := data.GetDeal(jobOffer.JobOffer, cheapestResourceOffer)
-			if err != nil {
-				span.SetStatus(codes.Error, "unable to get deal")
-				span.RecordError(err)
-				return nil, err
-			}
-			span.AddEvent("get_deal.done", trace.WithAttributes(attribute.String("deal.id", deal.ID)))
-
-			// add the match decision for this job offer
-			for _, matchingResourceOffer := range matchingResourceOffers {
-
-				addDealID := ""
-				if cheapestResourceOffer.ID == matchingResourceOffer.ID {
-					addDealID = deal.ID
+			var selectedResourceOffer data.ResourceOffer
+			if jobOffer.JobOffer.Mode == data.MarketPrice {
+				// For market price jobs, select the cheapest offer, breaking ties with age
+				selectedResourceOffer = matchingResourceOffers[0]
+				for _, offer := range matchingResourceOffers[1:] {
+					if isCheaperOrOlder(offer, selectedResourceOffer) {
+						selectedResourceOffer = offer
+					}
+				}
+			} else {
+				// For fixed price jobs, select the first (oldest) offer that meets the price requirement
+				for _, offer := range matchingResourceOffers {
+					if offer.DefaultPricing.InstructionPrice <= jobOffer.JobOffer.Pricing.InstructionPrice {
+						selectedResourceOffer = offer
+						break
+					}
 				}
+			}
 
-				span.AddEvent("add_match_decision.start")
-				_, err := db.AddMatchDecision(matchingResourceOffer.ID, jobOffer.ID, addDealID, true)
+			// Selected resource offer ID will be set when we selected a resource offer.
+			// Otherwise it will be a zero-value empty string.
+			if selectedResourceOffer.ID != "" {
+				span.AddEvent("get_deal.start",
+					trace.WithAttributes(
+						attribute.String("selected_resource_offer", selectedResourceOffer.ID),
+						attribute.KeyValue{
+							Key:   "matching_resource_offers",
+							Value: attribute.StringSliceValue(data.GetResourceOfferIDs(matchingResourceOffers)),
+						}))
+
+				// Make a deal with the job offer and selected resource offer
+				deal, err := data.GetDeal(jobOffer.JobOffer, selectedResourceOffer)
 				if err != nil {
-					span.SetStatus(codes.Error, "unable to add match decision")
+					span.SetStatus(codes.Error, "unable to get deal")
 					span.RecordError(err)
 					return nil, err
 				}
-				span.AddEvent("add_match_decision.done")
-			}
+				span.AddEvent("get_deal.done", trace.WithAttributes(attribute.String("deal.id", deal.ID)))
+
+				// Add match decisions for all matching offers
+				for _, matchingOffer := range matchingResourceOffers {
+					addDealID := ""
+					if selectedResourceOffer.ID == matchingOffer.ID {
+						addDealID = deal.ID
+					}
+
+					// All match decisions had matching resource offers, set the result to true.
+					// The match decision has a deal ID if it's resource offer was selected.
+					span.AddEvent("add_match_decision.start")
+					_, err := db.AddMatchDecision(matchingOffer.ID, jobOffer.ID, addDealID, true)
+					if err != nil {
+						span.SetStatus(codes.Error, "unable to add match decision")
```

Comment:
Should have a log an error log message here

Reply:
Done: c56ccfb70bc00117d0239c5130ee80d024a99af1